### ISSUE TITLE
Injecting an unbound NewType gives an unexpected TypeError

### ISSUE THREAD

**Issue by davidparsson** [alecthomas/injector#117]
See the following example:

```
String1 = NewType('String1', str)

class A:
    @injector.inject
    def __init__(self, string_1: String1) -> None:
        self.string_1 = string_1

i = injector.Injector()
a = i.create_object(A)  # This fails
```

The above code raises the following:
```
TypeError: Injecting partially applied functions is no longer supported.
```

To me that's a bit confusing since `NewType` is supported.

**Comment by jstasiak** [comment]
Can you provide full stack trace? Once I'm done with my other refactoring I'll have a look at this, should be rather simple to fix.

**Comment by davidparsson** [comment]
Here you go:

```
>>> a = i.create_object(A)

Traceback (most recent call last):
  File "/.../site-packages/injector/__init__.py", line 468, in get_binding
    return self._get_binding(key, only_this_binder=is_scope)
  File "/.../site-packages/injector/__init__.py", line 463, in _get_binding
    raise KeyError
KeyError
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/.../site-packages/injector/__init__.py", line 770, in create_object
    (), additional_kwargs, e, self._stack,)
  File "/.../site-packages/injector/__init__.py", line 67, in reraise
    raise exception.with_traceback(tb)
  File "/.../site-packages/injector/__init__.py", line 763, in create_object
    init(instance, **additional_kwargs)  # type: ignore # python/mypy/issues/4001
  File "/.../site-packages/injector/__init__.py", line 1109, in inject
    kwargs=kwargs
  File "/.../site-packages/injector/__init__.py", line 858, in call_with_injection
    owner_key=self_.__class__ if self_ is not None else callable.__module__,
  File "/.../site-packages/injector/__init__.py", line 55, in wrapper
    return function(*args, **kwargs)
  File "/.../site-packages/injector/__init__.py", line 901, in args_to_inject
    instance = self.get(key.interface)
  File "/.../site-packages/injector/__init__.py", line 719, in get
    binding, binder = self.binder.get_binding(None, key)
  File "/.../site-packages/injector/__init__.py", line 477, in get_binding
    binding = self.create_binding(key.interface)
  File "/.../site-packages/injector/__init__.py", line 400, in create_binding
    provider = self.provider_for(interface, to)
  File "/.../site-packages/injector/__init__.py", line 451, in provider_for
    raise TypeError('Injecting partially applied functions is no longer supported.')
TypeError: Injecting partially applied functions is no longer supported.
```

**Comment by jstasiak** [comment]
Fixed in just release 0.17.0.
